### ISSUE TITLE
Matrix as grid with changing central meridian doesn't work well for gridline grids

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#3844]
**Full script that generated the error**

The script generates 4 maps, for pixel or gridline grids passing as grid or matrix:
```python
import pygmt
from pygmt.datasets import load_earth_relief
import xarray as xr

# pixel grid
fig = pygmt.Figure()
fig.grdimage("@earth_relief_01d_p", projection="W120/15c", cmap="geo")
fig.savefig("grid_p.png")

# pixel matrix
grid = load_earth_relief(registration="pixel")
fig = pygmt.Figure()
fig.grdimage(grid, projection="W120/15c", cmap="geo")
fig.savefig("matrix_p.png")


# gridline grid
fig = pygmt.Figure()
fig.grdimage("@earth_relief_01d_g", projection="W120/15c", cmap="geo")
fig.savefig("grid_g.png")

# gridline matrix
grid = load_earth_relief(registration="gridline")
fig = pygmt.Figure()
fig.grdimage(grid, projection="W120/15c", cmap="geo")
fig.savefig("matrix_g.png")
```

Use the command below to compare the two maps of pixel-registered grid/matrix. The RMS is zero. So passing grid or matrix give the same results:
```
gm compare -density 200 -highlight-color magenta -highlight-style assign -metric rmse -file diff_p.png matrix_p.png grid_p.png
```

Use the command below to compare the two maps of pixel-registered grid/matrix.
```
gm compare -density 200 -highlight-color magenta -highlight-style assign -metric rmse -file diff_g.png matrix_g.png grid_g.png
```
The RMS is non-zero. Here is the diff image. There are tiny difference along the x=180 line. Perhaps a 0.5 grid offset?

![image](https://user-images.githubusercontent.com/3974108/89378026-f9215e80-d6c0-11ea-8540-5c06bc03c02a.png)

**Comment by PaulWessel** [comment]
Will look tomorrow.  I know the issue: When we rotate the central longitude the repeated columns at -180 and 180 need to collapse to one.

**Comment by PaulWessel** [comment]
So 1-2 days go by, an automatic OS update happened over night, so GMT_LIBRARY_PATH was unset.  However, I am missing another step before I can sy import pygmt, something about switching from base to pygmt or a setup step?

**Comment by seisman** [comment]
Yes, you need to activate your conda environment by `conda activate pygmt`.

**Comment by PaulWessel** [comment]
??? My outputs look like this (with master):

![matrix_g](https://user-images.githubusercontent.com/26473567/89465479-e2edbf80-d70d-11ea-96be-71addd8190be.png)

**Comment by seisman** [comment]
The script generates 4 figures. Which one?

**Comment by PaulWessel** [comment]
They are all like that (just a sliver near the boundary), I think this one was matrix_g.png

**Comment by seisman** [comment]
This makes no sense. Perhaps try `pygmt.show_versions()` and see if you're using the correct GMT or PyGMT version?

**Comment by PaulWessel** [comment]
Makes perfect sense.  There was an old gmt.history in the directory I started python:

# GMT 6 Session common arguments shelf
BEGIN GMT 6.2.0
B       0g10
J       M
JM      M15c
JQ      Q15/6i
R       -115/-100/55/60
@L      1
END

and the python script latched onto that.  SHould not pygmt start a new session and no gmt.history is used?

**Comment by seisman** [comment]
You mean the **gmt begin -C**? 

> Start this session with a clean slate: Any gmt.conf files in the usual search path directories are ignored [Default starts session with the prevailing user settings].

Does it also ignore gmt.history?

**Comment by PaulWessel** [comment]
So back to the original issue: Since we are passing reference we simply use the grid.  It gets projected, in gmt_grd_project there will be loops over input and output nodes (you remember this from last week).  So there will be two sets of meridians taht are used (lon = -180 and lon = 180) which are exactly the same.  So those should technically be given a 0.5 weight so that in averaging many nodes they should not count more.  Normally (i.e., grdfiles) we do not see this (or possibly never noticed) because we read int the grid with rotation, so the repeated columns are at the two separate sides.  Now they show up in the middle.

I will see if ti is possibly to insert the weight of 0.5.

**Comment by PaulWessel** [comment]
If you start with begin -C then clearly we leak that through.  Give it a test bu using my history file above and confirm the problem.

**Comment by PaulWessel** [comment]
Testing a fix for this.  Now it passes:
           Normalized    Absolute
          ============  ==========
     Red: 0.0054440957        1.4
   Green: 0.0035965496        0.9
    Blue: 0.0052102560        1.3
   Total: 0.0048207923        1.2

versus before it was

           Normalized    Absolute
          ============  ==========
     Red: 0.0076255501        1.9
   Green: 0.0050802491        1.3
    Blue: 0.0066723388        1.7
   Total: 0.0065441590        1.7

All tests still pass so I am making a PR.

**Comment by PaulWessel** [comment]
Since the history file is read in GMT_Create_Session, it has already been ingested when begin -C is parsed. Thus, I need to reset the history.  I could not recreate the problem on the command line, so suspect it is interfering with the matri/grdimage stuff.

**Comment by PaulWessel** [comment]
There should be no instances where an old gmt.history file should be used by a new modern script, -C or not.  I think I will always reset history when gmt begin is called.  It is different for the settings.

**Comment by seisman** [comment]
Closed by #3849.
